CMakeLists.txt

```diff
@@ -31,6 +31,8 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 include(CheckIncludeFileCXX)
 check_include_file_cxx(filesystem HAS_FILESYSTEM)
 
+option(OPENBLACK_WARNINGS_AS_ERRORS "Treat warnings as errors (disable easier prototyping, enable for CI)" OFF)
+
```

Comment:
My question to the reviewer is should this be on by default and the dev turns it off themselves or leave it off by default?

Reply:
Leave it off by default definitely.